openmc/mesh.py

```diff
@@ -1315,6 +1336,12 @@ def write_data_to_vtk(self, filename, datasets, volume_normalization=True):
         pts_cartesian[:, 0] = r * np.cos(phi)
         pts_cartesian[:, 1] = r * np.sin(phi)
 
+        # offset with origin
+        if any([coord != 0 for coord in self.origin]):
+            pts_cartesian[:, 0] = pts_cartesian[:, 0] + self.origin[0]
+            pts_cartesian[:, 1] = pts_cartesian[:, 1] + self.origin[1]
```

Comment:
I think just add the origin coordinates in lines 1336 and 1337 instead of bothering to check

Reply:
I tend to agree, but @pshriwise wanted to keep this check. Patrick was it for performance issues?

> Yeah I suppose it's more explicit to return `(0.0, 0.0, 0.0)` than `None`. This is true.
> 
> I guess I'd still like to make sure we're skipping that translation step when it isn't needed. NumPy might be smart enough to recognize that as a no-op, but I'm not sure so some clarity on that would be good. If it isn't going to skip it then I'll suggest that we keep a check for `self.origin != (0.0, 0.0, 0.0)` in that translation block.

Reply:
I don't think this is a big deal from a performance perspective, but I'll try to track down proof. I also don't like using `!=` with floats. So, to avoid these comparisons or using `np.isclose` and specifying a tolerance I would say the clearest/cleanest approach is to just to always add them.

Reply:
Yeah, true. Good to avoid floating point comparisons if possible. Overall, I'd probably prefer this property be set to `None` by default instead of `(0.0, 0.0, 0.0)`.

Reply:
> Yeah I guess I don't know enough to know if NumPy will skip this or if we'll be adding a bunch of zeros to our points. @eepeterson do you know?
> 
> I'll admit it's cleaner without this check..

I don't think NumPy will skip this. I think it will do the addition, but I don't think there will be any significant overhead if it is added to the coordinate transformation line. I think the `r*np.cos(phi)` portion is much more expensive for large arrays, but we have to do it. FWIW `pts_cartesian[:, 2] += self.origin[2]` actually seems to be ~2x faster for large arrays (testing with 1e8 x 3 array).

Reply:
Sorry no in comparison to `pts_cartesian[:, 2] = pts_cartesian[:, 2] + self.origin[2]`

Reply:
Oh! Yeah the in-place operator should be faster here. Nice catch

Reply:
Also on line 1334 if you are going to end up filling the whole `pts_cartesian` array anyway you don't need to copy the cylindrical points values just use `np.empty_like(pts_cylindrical)` instead, it's much faster. Same goes for line 1594.